pkg/upgraders/upgradeable.go

```diff
@@ -0,0 +1,74 @@
+package upgraders
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/blang/semver"
+	"github.com/go-logr/logr"
+	configv1 "github.com/openshift/api/config/v1"
+	upgradev1alpha1 "github.com/openshift/managed-upgrade-operator/pkg/apis/upgrade/v1alpha1"
+	cv "github.com/openshift/managed-upgrade-operator/pkg/clusterversion"
+)
+
+func (c *clusterUpgrader) IsUpgradeable(ctx context.Context, logger logr.Logger) (bool, error) {
+	cV := &configv1.ClusterVersion{}
+	uC := &upgradev1alpha1.UpgradeConfig{}
+
+	//Check if the cluster is already upgrading -
+	upgradeCommenced, err := c.cvClient.HasUpgradeCommenced(c.upgradeConfig)
+	if err != nil {
+		return false, err
+	}
+	if upgradeCommenced {
+		logger.Info(fmt.Sprintf("Skipping upgrade step %s", upgradev1alpha1.UpgradePreHealthCheck))
+		return true, nil
+	}
+
+	// get current clusterversion
+	cvVersion, err := cv.GetCurrentVersion(cV)
+	if err != nil {
+		logger.Error(err, "failed to get current cluster version")
+		return false, err
+	}
+
+	parsedCvVersion, err := semver.Parse(cvVersion)
+	if err != nil {
+		logger.Error(err, fmt.Sprintf("Failed to parse current cluster version %s as semver", cvVersion))
+		return false, err
+	}
+
+	// get the desired clusterversion from upgradeconfig
+	ucVersion := uC.Spec.Desired.Version
+	if err != nil {
+		logger.Error(err, "failed to get current upgrade config desired version")
+		return false, err
+	}
+
+	parsedUcVersion, err := semver.Parse(ucVersion)
+	if err != nil {
+		logger.Error(err, fmt.Sprintf("failed to parse upgrade config desired version %s as semver", ucVersion))
+		return false, err
+	}
+
+	// check if the clusterversion has "Upgradeable" else return true
+	for _, val := range cV.Status.Conditions {
+		if val.Type != configv1.OperatorUpgradeable {
+			return true, nil
+		}
+	}
```

Comment:
In its current form, this is going to return that the cluster is upgradeable on the very first condition it sees which has a type which is not "upgradeable". 

What we want to happen here is to check if there is no Upgradeable condition type at all. If that is the case, then we're not on a version of OCP where we can even look at Upgradeable / ack gates, so we should default to returning true.

So you'll want something like this (note, I have not tested this code):
```suggestion
	// check if the clusterversion has "Upgradeable" else return true
	hasUpgradeableCondition := false
	for _, val := range cV.Status.Conditions {
		if val.Type == configv1.OperatorUpgradeable {
			hasUpgradeableCondition = true
			break
		}
	}
	if !hasUpgradeableCondition {
	    // CVO has no Upgradeable condition to gate upgrades, so we can infer the cluster is upgradeable
	    return true, nil
	}
```